src/entity/user/UserTransaction.ts

```diff
@@ -0,0 +1,77 @@
+import {
+  Column,
+  CreateDateColumn,
+  Entity,
+  JoinColumn,
+  ManyToOne,
+  PrimaryGeneratedColumn,
+  UpdateDateColumn,
+} from 'typeorm';
+import type { User } from './User';
+import type { Product } from '../Product';
+import type { RequestMeta } from '../../Context';
+
+export type UserTransactionFlags = Partial<{
+  note: string;
+}>;
+
+export type UserTransactionFlagsPublic = Pick<UserTransactionFlags, 'note'>;
+
+export type UserTransactionRequest = RequestMeta;
+
+@Entity()
+export class UserTransaction {
+  @PrimaryGeneratedColumn('uuid')
+  id: string;
+
+  @Column({ type: 'uuid', nullable: true })
+  productId: string | null;
+
+  @ManyToOne('Product', {
+    lazy: true,
+    onDelete: 'CASCADE',
+  })
+  product: Promise<Product>;
+
+  @Column()
+  status: number; // TODO feat/transactions enum from schema later
+
+  @CreateDateColumn()
+  createdAt: Date;
+
+  @UpdateDateColumn()
+  updatedAt: Date;
+
+  @Column()
+  receiverId: string;
+
+  @ManyToOne('User', {
+    lazy: true,
+    onDelete: 'CASCADE',
+  })
+  @JoinColumn({ name: 'receiverId' })
+  receiver: Promise<User>;
+
+  @Column({ nullable: true })
+  senderId: string | null;
+
+  @ManyToOne('User', {
+    lazy: true,
+    onDelete: 'CASCADE',
+    nullable: true,
+  })
+  @JoinColumn({ name: 'senderId' })
+  sender: Promise<User>;
+
+  @Column()
+  value: number;
+
+  @Column()
+  fee: number;
```

Comment:
Who will calculate this now?

Reply:
This will be a flat fee from remote config at the time of making a transaction, so no calculations, just a way of knowing what was sent to njord

Reply:
Ah ok, so remote config is in charge of fees then?

Reply:
it is in charge of getting fixed fee percentage and sending it to njord, everything else is on njord side eg. how much credits will be deducated, given etc..